ext/standard/php_uuid.h

```diff
@@ -0,0 +1,163 @@
+/*
+  +----------------------------------------------------------------------+
+  | PHP Version 7                                                        |
+  +----------------------------------------------------------------------+
+  | Copyright (c) 1997-2017 The PHP Group                                |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | http://www.php.net/license/3_01.txt                                  |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Richard Fussenegger <php@fleshgrinder.com>                   |
+  +----------------------------------------------------------------------+
+*/
+
+/* $Id$ */
+
+#ifndef PHP_UUID_H
+#define PHP_UUID_H
+
+#include "zend_types.h"
+#include "php.h"
+
+PHPAPI extern zend_class_entry *php_ce_UUID;
+PHPAPI extern zend_class_entry *php_ce_UUIDParseException;
+
+#define PHP_UUID_LEN 16
+typedef struct php_uuid {
+	uint8_t bytes[PHP_UUID_LEN];
+} php_uuid;
+
+#define PHP_UUID_HEX_LEN 33
```

Comment:
I'd recommend using the value 32 here and writing PHP_UUID_HEX_LEN+1 in the struct below. By convention the term "length" does not include the null byte. If you want to keep the value, then the constant name should be PHP_UUID_HEX_SIZE.

Reply:
Or better: `#define PHP_UUID_HEX_LEN (PHP_UUID_LEN*2)`